contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/VoldemortSwapJob.java

```diff
@@ -89,6 +101,89 @@ public VoldemortSwapJob(String id,
+        EmbeddedGobblin embeddedGobblin = new EmbeddedGobblinDistcp(from, to).mrMode();
+
+        // Used for global throttling"
+        embeddedGobblin.distributeJar("lib/*");
+
+        for (Map.Entry<String, String> entry : this.props.entrySet()) {
+            if (entry.getKey() instanceof String && ((String) entry.getKey()).startsWith("distcpConf.")) {
+                String key = ((String) entry.getKey()).substring("distcpConf.".length());
+                embeddedGobblin.setConfiguration(key, entry.getValue().toString());
+            }
+        }
+        JobExecutionResult result =  embeddedGobblin.run();
+
+        if (!result.isSuccessful()) {
+            throw new RuntimeException("Distcp job failed!", result.getErrorCause());
+        }
+    }
+
+    private FileSystem getRootFS(String target) throws Exception {
+        URI rootURI = new URI(target.replaceAll("(?<=:[0-9]{1,5})/.*", ""));
+        FileSystem rootFS = FileSystem.get(rootURI, new JobConf());
+        return rootFS;
+    }
+
+    private void deleteDir(FileSystem fs, String target) throws Exception {
+        Path path = new Path(target.replaceAll(".*://.*?(?=/)", ""));
+
+        if (fs.exists(path)) {
+            fs.delete(path, true);
+            if (fs.exists(path)) {
+                warn("Could not delete temp directory " + path + " in CDN!");
+            } else {
+                info("Deleted " + path);
+            }
+        }
+    }
+
+    private void deleteDirOnExit(FileSystem fs, String target) throws Exception {
+        Path path = new Path(target.replaceAll(".*://.*?(?=/)", ""));
+
+        fs.deleteOnExit(path);  // Delete the directory even if an exception occurs
+        info(path + " is scheduled to be deleted on exit.");
+    }
+
+    private void addPermissionsToParents(FileSystem fs, String target) throws Exception {
+        String pathFromRoot = target.replaceAll(".*://.*?/", "");
+        String parent = "";
+
+        for (String seg: pathFromRoot.split("/")) {
+            parent = parent + "/" + seg;
+            Path parentPath = new Path(parent);
+            FsPermission perm = fs.getFileStatus(parentPath).getPermission();
+            if (!perm.getOtherAction().implies(FsAction.READ_EXECUTE)) {
+                fs.setPermission(parentPath, new FsPermission(perm.getUserAction(), perm.getGroupAction(), perm.getOtherAction().or(FsAction.READ_EXECUTE)));
+                info("for path " + parent + ", permissions changed:");
+                info("from: " + perm.toString());
+                info("  to: " + fs.getFileStatus(parentPath).getPermission().toString());
+            }
+        }
+    }
+
+    private String pickCDN() {
+        List<String> pushClusters = props.getList(VoldemortBuildAndPushJob.PUSH_CLUSTER);
+
+        if (!props.containsKey(VoldemortBuildAndPushJob.PUSH_CDN_CLUSTER)) {
+            return "";
+        }
+        List<String> cdnClusters = props.getList(VoldemortBuildAndPushJob.PUSH_CDN_CLUSTER);
```

Comment:
This 'cdnClusters' is a little confusing, so could we maintain a map between the cluster name and the corresponding cdn url?